### PR TITLE
Change SF deploy to use certificate from key vault

### PR DESCRIPTION
**Pull request by alexperovich**
https://github.com/dotnet/core-eng/issues/10696

### CODE REVIEW

File: eng/deploy.yaml [hunk 1]
```diff
@@ -111,22 +111,30 @@ stages:
       artifact: MaestroApplication
     - download: current
       artifact: ReleaseUtilities
-    - task: ServiceFabricPowerShell@1
-      displayName: Refresh Service Fabric Services (Maestro)
+    - task: AzureCLI@2
+      displayName: Deploy Service Fabric Application
```
Comment (sunandabalu) [inline eng/deploy.yaml:115]: probably good idea to add Meastro somewhere in the display name

**Review by sunandabalu — APPROVED** [review]

**Comment by sunandabalu** [comment]
Could you attach an associated issue to the PR?

**Comment by JohnTortugo** [comment]
@alexperovich - LGTM but I left the team a few months ago .. :-)

**Comment by alexperovich** [comment]
O yea, I just clicked all the suggestions. :)

**Comment by riarenas** [comment]
@alexperovich This seems to have broken the deployment: https://dnceng.visualstudio.com/internal/_build/results?buildId=856530&view=logs&j=b277f1ae-44e7-5ce4-4818-00049b2c7654&t=d8aeea63-ad78-5c06-b508-c02d12fd77d5

```

Get-Content : Cannot find path 'D:\a\1\s\Int' because it does not exist.
At D:\a\1\s\eng\deployment\deploy.ps1:12 char:29
+ $publishProfileXml = [Xml] (Get-Content $PublishProfile)
+                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (D:\a\1\s\Int:String) [Get-Content], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand
 
##[error]Script failed with exit code: 1
```

File: eng/deploy.yaml [hunk 2]
```diff
@@ -111,22 +111,30 @@ stages:
...
+            "ApplicationName" = "fabric:/MaestroApplication";
+            "ApplicationManifestPath" = $env:ApplicationManifestPath;
+            "ServicesSourceFolder" = $env:ServicesSourceFolder;
+          }
+          if ($env:DisableAutoRollBack -eq 'true') {
+            $parameters["ForceUpgrade"] = $true;
+          }
+          eng/deployment/deploy.ps1 @parameters
+      env:
+        PublishProfile: $(Pipeline.Workspace)/MaestroApplication/projectartifacts/PublishProfiles/$(PublishProfile).xml
```
Comment (riarenas) [inline eng/deploy.yaml:134]: Looks like this is ending up as `D:\a\1\s\Int` in the build machines and that's resulting in a file not found error?

Comment (alexperovich) [reply eng/deploy.yaml:134]: Yea, https://github.com/dotnet/arcade-services/pull/1444

Comment (riarenas) [reply eng/deploy.yaml:134]: Yeah I don't know how I missed that since Friday. Sorry 🙊

**Pull request merged**